brave/brave-core
# Display IPFS interstitial page when failed to use local node

Resolves https://github.com/brave/brave-browser/issues/11558

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:
1. Make sure IPFS feature flag is enabled.
2. Go to brave://settings and search for ipfs, make sure local node is selected and check the new switch should be disabled at the first place.
<img width="682" alt="Screen Shot 2020-10-06 at 4 42 13 PM" src="https://user-images.githubusercontent.com/4730197/95271628-32766780-07f3-11eb-9213-d7b9fdf2f069.png">

3. Turn off your network and visit http://127.0.0.1:8080/ipfs/QmbWqxBEKC3P8tqsKc98xmWNzrzDtRLMiMPL8wBuTGsMnR
4. An interstitial page should be shown
<img width="1166" alt="Screen Shot 2020-10-06 at 4 41 42 PM" src="https://user-images.githubusercontent.com/4730197/95271804-a4e74780-07f3-11eb-83b7-cfef57a92850.png">

5. Go to brave://ipfs, verified that daemon was launched with 0 connected peers.
<img width="1168" alt="Screen Shot 2020-10-06 at 4 41 47 PM" src="https://user-images.githubusercontent.com/4730197/95271745-84b78880-07f3-11eb-8017-fb7181201d74.png">

6. Turn your network back on and go click the proceed button in the interstitial page.
7. The page should be redirected to public gateway
<img width="1163" alt="Screen Shot 2020-10-06 at 4 48 40 PM" src="https://user-images.githubusercontent.com/4730197/95271878-e1b33e80-07f3-11eb-9d62-94f75f52f9ec.png">

8. Go the brave://settings and search for ipfs, the new gateway fallback switch should now be enabled.

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
